Sources/Drill/Invocation.swift

```diff
@@ -46,61 +58,64 @@ public struct Invocation {
       return true
     }
 
+    let context = PassContext(engine: engine)
+
+    defer {
+      if options.shouldPrintTiming {
+        context.timer.dump(to: &stdoutStream)
+      }
+    }
+
+    let shineFile = Passes.lex |> Passes.shine
+    let parseFile = shineFile |> Passes.parse
+    let scopeCheckFile =
+      parseFile |> Passes.scopeCheck
+
     for path in sourceFiles {
       let url = URL(fileURLWithPath: path)
-      let contents = try String(contentsOf: url, encoding: .utf8)
-      let lexer = Lexer(input: contents, filePath: path)
-      let tokens = lexer.tokenize()
+
+      func run<PassTy: PassProtocol>(_ pass: PassTy) -> PassTy.Output?
+        where PassTy.Input == URL {
```

Comment:
Nit: Is that the style here?  I always indent the next brace

```
func run<PassTy: PassProtocol>(_ pass: PassTy) -> PassTy.Output?
   where PassTy.Input == URL 
{
```

Reply:
Yeah, I can put it on a new line 👍